### PR TITLE
Migrate Travis CI builds to container-based infrastructure (#235)

### PR DESCRIPTION
**Pull request by xabolcs**
Easy transition for #235.

@whimboo please review!

### CODE REVIEW

**Comment by xabolcs** [comment]
Ok, this could wait a few days or more!

**Comment by xabolcs** [comment]
Now it looks promising .. this PR now got the same error as in #234! :+1:

File: .travis.yml [hunk 1]
```diff
@@ -10,12 +15,14 @@ env:
...
+addons:
+  apt:
+    packages:
+    - ant
+    - python-pip
 
 install:
-  - sudo apt-get install -y ant python-pip
-  - sudo pip install mozdownload==1.12
+  - pip install --user mozdownload==1.12
```
Comment (whimboo) [inline .travis.yml]: oh wow. please change that to `mozdownload==1.19`! 1.12 will definitely fail to download the Firefox build.

Comment (whimboo) [inline .travis.yml]: Maybe make it even without a fixed version so we always use the latest version for testing.

Comment (xabolcs) [inline .travis.yml]: That change is covered in PR #237.

**Comment by xabolcs** [comment]
Ok, this looks good now! :)

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,10 @@
+sudo: false
+
 language: node_js
 
+node_js:
+  - "node"
```
Comment (whimboo) [inline .travis.yml]: Mind telling me why we explicitly have to specify it?

Comment (xabolcs) [inline .travis.yml]: Without that the build wasn't able to step forward from `nvm install 0.10`.
See Travis CI [build #173][1]!

[1]: https://travis-ci.org/mozilla/memchaser/builds/89566396

Comment (whimboo) [inline .travis.yml]: Ok, so a version has to be given here. And using "node" will pick the latest release. I think that this is fine.

**Comment by whimboo** [comment]
Looks fine to me. I don't think we need a cache for the memchaser build. So please get it merged when you have the time.

And thanks for working on it!

**Comment by xabolcs** [comment]
Thanks! Will do it soon!

**Comment by xabolcs** [comment]
Landed as https://github.com/mozilla/memchaser/commit/beb18f0e842892d195d475ab26459e2ade183305.

**Pull request closed without merging**